src/main/java/org/skriptlang/skript/lang/util/SkriptQueue.java

```diff
@@ -0,0 +1,85 @@
+package org.skriptlang.skript.lang.util;
+
+import ch.njol.yggdrasil.YggdrasilSerializable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+
+/**
+ * A queue of elements.
+ * This is designed not to support null elements, i.e. putting in nothing should do nothing.
+ */
+public class SkriptQueue extends LinkedList<@NotNull Object>
```

Comment:
We don't have any official support for off-thread things in general. I don't like adding in that kind of patchwork safety because it gives the illusion that it's okay to use in those situations, when we actually have no idea.

Reply:
We do through async events / async/parallel loading, though they are not yet fully functional, so we can always revisit this.